File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        val=n
        sum,prod=0,1
        while(val>0):
            v=int(val%10)
            val/=10
            val=int(val)
            sum+=v
            prod*=v
        sum+=prod
        return n%sum==0

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        String str=Integer.toString(n);
        int prod=1, sum=0;
        for(int i=0; i<str.length(); i++){
            int val=str.charAt(i)-'0';
            prod*=val;
            sum+=val;
        }
        sum+=prod;
        return n%sum==0;
    }
}